base/dict.jl

```diff
@@ -698,6 +706,11 @@ function get{K,V}(default::Callable, h::Dict{K,V}, key)
     return (index < 0) ? default() : h.vals[index]::V
 end
 
+function tryget{K,V}(h::Dict{K,V}, key)
+    index = ht_keyindex(h, key)
+    return (index<0) ? Nullable{V}() : Nullable{V}(h.vals[index]::V)
```

Comment:
is this more efficient than the abstract definition above? the generic one should be correct here right?